[TensorRT EP] update oss parser to latest

yf711 opened the pull request:
### Description
<!-- Describe your changes. -->
* Update oss parser version to latest commit of 10.8-GA branch

### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here. -->

* Action needed to adapt latest onnx-tensorrt 10.8-GA branch to fix scatterND attribute issue and `plugin.h` not found issue

chilo-ms commented:
The TRT CI has building issue.

yf711 commented:
onnx-tensorrt introduced a dependency `plugin.h`, which is available by tarball/zip installation (under`TensorRT-tarball/python/include/impl/`) but not available by linux package installation. Windows CIs are not affected

jywu-msft commented:
this is a bit confusing. if we publish packages built against TRT 10.8 but stay with10.7 parser?

chilo-ms commented:
> onnx-tensorrt introduced a dependency [plugin.h](https://github.com/onnx/onnx-tensorrt/blob/10.8-GA/CMakeLists.txt#L109-L113), which is available by tarball/zip installation (underTensorRT-tarball/python/include/impl/) but not available by linux package installation. OSS parser users who use Windows/Linux tarball installation are not affected.
Updating the logic to make sure builtin parser users are not blocked
Why previously when our CI is running with TRT 10.6 built-in parser didn't complain about this plugin.h depedency?

yf711 commented:
> > onnx-tensorrt introduced a dependency [plugin.h](https://github.com/onnx/onnx-tensorrt/blob/10.8-GA/CMakeLists.txt#L109-L113), which is available by tarball/zip installation (underTensorRT-tarball/python/include/impl/) but not available by linux package installation. OSS parser users who use Windows/Linux tarball installation are not affected.
> > Updating the logic to make sure builtin parser users are not blocked
> 
> Why previously when our CI was running with TRT 10.6 built-in parser didn't complain about this plugin.h depedency?

This dependency is only needed by oss parser. CIs that use Bulit-in parser have no issue even though there's no plugin.h

chilo-ms commented:
> Updating the logic to make sure builtin parser users are not blocked

You mention "Updating the logic to make sure builtin parser users are not blocked", it should be oss parser, right?

chilo-ms commented on cmake/onnxruntime_providers_tensorrt.cmake:
```diff
@@ -57,16 +56,32 @@
-  endif()
-
+  # Check TRT version >= requirement
+  function(check_trt_version req_major req_minor req_patch out_var)
```
This function is much complex than the original check.
Any reasons to add this function?

yf711 replied:
So far it's not needed. I will clean this

yf711 commented:
> > Updating the logic to make sure builtin parser users are not blocked
> 
> You mention "Updating the logic to make sure builtin parser users are not blocked", it should be oss parser, right?

I didn't mention that clearly. 
The original code I added (search plugin.h without if check) was a blocker. This part is now removed, as it's not needed.
